src/ncp-spinel/SpinelNCPInstance-DataPump.cpp

```diff
@@ -247,7 +247,41 @@ SpinelNCPInstance::ncp_to_driver_pump()
 		{
 			uint16_t frame_crc = (mInboundFrame[mInboundFrameSize]|(mInboundFrame[mInboundFrameSize+1]<<8));
 			if (mInboundFrameHDLCCRC != frame_crc) {
+
+				int i;
+				static const uint8_t kAsciiCR = 13;
+				static const uint8_t kAsciiBEL = 7;
+
 				syslog(LOG_ERR, "[NCP->]: Frame CRC Mismatch: Calc:0x%04X != Frame:0x%04X, Garbage on line?", mInboundFrameHDLCCRC, frame_crc);
+
+				// This frame might be an ASCII backtrace, so we check to
+				// see if all of the characters are ascii characters, and if
+				// so we dump out this packet directly to syslog.
+
+				mInboundFrameSize += 2;
+
+				for (i = 0; i < mInboundFrameSize; i++) {
+					// Acceptable control codes
```

Comment:
subjective style nit: I tend to like to have a temporary for accesses into an array. I.e:
`uint8_t currentByte = mInboundFrame[i];`

It's all the same to the compiler but I've always been found it more readable